pkg/operation/botanist/addons.go

```diff
@@ -398,6 +405,17 @@ func (b *Botanist) generateCoreAddonsChart() (*chartrenderer.RenderedChart, erro
 		networkPolicyConfig.NodeLocalDNS.KubeDNSClusterIP = NodeLocalIPVSAddress
 	}
 
+	if b.APIServerSNIEnabled() {
+		coreDNSConfig["kubeAPIServerHost"] = kasFQDN
+		metricsServerConfig["kubeAPIServerHost"] = kasFQDN
+		nodeProblemDetectorConfig["env"] = []interface{}{
+			map[string]interface{}{
+				"name":  "KUBERNETES_SERVICE_HOST",
+				"value": "kasHost",
```

Comment:
```suggestion
				"value": b.outOfClusterAPIServerFQDN(),
```

Reply:
I fixed it before creating the PR, but pushed the change to the wrong git remote... Have a look at 009a34b